### ISSUE TITLE
compileString on browser with @use './src/index.scss'

### ISSUE THREAD

**Issue by aamirtestable** [sass/sass#3986]
I am playing around with compileString and thought to give a playground / storybook type thing to other developers what kind of changes i'm making. I have created an html page with vite and started building a theme for the company. 

Now in the playground, scss is working fine, just that i'm not sure how to import an scss file. which have  something like 

```scss
// /src/index.scss
@forward 'components/crm-badge';` // this have a mixin of `badge`
```

And the tools scss will be like this

```scss
@use `./src/index` as UI;

.contact_logo{
   @include UI.logo
   & > .badge {
      @include UI.badge
   }
}
```

So far my app and scss integrations all are working fine, just have an itch to do that in web browser 😸 
Can I load scss over the server and its all `forwards`? 

---- edit

Ahh one more thing, i just realized and this can be an overkill but might be good.  Taking the same example from top. I have created 3 panels. one is visual output how the badge would look, and on right side top editor where that code snippet will be, and in the right side bottom part compiled css. 

I believe that compiled css would include everything like my other components like charts and reports. 

So may be only show following in compiled one

```css
.contact_logo{
   background-color: #ddd;
   border-radius: 4px;
}

.contact_logo > .badge{
   border-radius: 4px;
   position: absolute;
   right: -5px;
   top: -5px;
   ...
}
```

In short just to show what only this component output will be instead of whole css file.

**Comment by nex3** [comment]
Although none of the default importers will work in a browser context, you can use the [importer API](https://sass-lang.com/documentation/js-api/interfaces/importer/) to support loads using custom semantics, whether that's loading over HTTP or through some sort of virtual filesystem.

Sass output is always a single CSS file, but you could theoretically use source maps to try to isolate the section that comes from a particular file.

**Comment by aamirtestable** [comment]
@nex3 

I have managed to run this and it loads all the files properly. Thank you

kind a stuck in a problem. It gives me this error. I can fix it by using `red($color)` but then in CLI when making build i got deprecations or `Expected color.red instead of red (scss/no-global-function-names)Stylelint[scss/no-global-function-names](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/no-global-function-names)`

```scss
Error: Undefined function.
   ╷
31 │     bg-rgb: #{color.channel($color, 'red')}, #{color.channel($color, 'green')}, #{color.channel($color, 'blue')};
   │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   ╵
```

right now I have changed it to red,green,blue 

Other thing I noticed that on each time i compile, it loads all the scss files again. Cant we just cache it somehow? 🤔

**Comment by nex3** [comment]
> kind a stuck in a problem. It gives me this error. I can fix it by using `red($color)` but then in CLI when making build i got deprecations or `Expected color.red instead of red (scss/no-global-function-names)Stylelint[scss/no-global-function-names](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/no-global-function-names)`
> 
> ```css-scss
> Error: Undefined function.
>    ╷
> 31 │     bg-rgb: #{color.channel($color, 'red')}, #{color.channel($color, 'green')}, #{color.channel($color, 'blue')};
>    │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>    ╵
> ```

It looks like whatever version of Sass you're using to compile in the browser is older than what you're using on the CLI, and doesn't support color CSS 4 colors. Make sure you're using the latest Sass version everywhere.

> Other thing I noticed that on each time i compile, it loads all the scss files again. Cant we just cache it somehow? 🤔

You can cache in your custom importer if you'd like. The Sass JS API doesn't currently support caching files across compilations (see https://github.com/sass/sass/issues/2745).

**Comment by aamirtestable** [comment]
Thank you
